README.md

```diff
@@ -83,9 +57,72 @@ $ cargo build --locked
 ```
 $ cargo test --locked
 ```
+## Usage
+
+First, generate a RISC-U binary (full support of RISC-V is coming soon!). Refer to the [selfie repositorie](https://github.com/cksystemsteaching/selfie). Once selfie is installed you can generate a binary for a code you have written in a file (e.g `hello.c`):
+
+```
+selfie -c hello.c -o hello.m
+```
+
+To display the available subcommands that Unicorn has you can type `./target/debug/unicorn --help`, or to display subcommands options `./target/debug/unicorn <SUBCOMMAND> --help`.
+
+Currently, we have 2 main options:
+### 1. Generate the BTOR2 file from the binary
+```
+./target/debug/unicorn beator hello.m --unroll 100 --solver boolector --out hello.btor2
+```
+The above command generate a BTOR2 file (i.e hello.btor2), while the unroll option especifies how many machine instruction (state transitions) Unicorn should represent in the BTOR2 file. In this example, Unicorn uses boolector to optimize at word level the number of variables the models needs. 
+
+There are more options. For example, you can pass the option `--bitblast`, and instead the BTOR2 file will represent a logic (combinatorial) circuit.
+
+
+### 2. Generate and/or test a QUBO of the binary
+```
+./target/debug/unicorn qubot hello.m --unroll 100 --solver boolector --out hello.unicorn --inputs 42,32;34
+```
+This command instead generate a qubo model that represent 100 executed instructions, and outputs a file describing the qubo in `hello.unicorn`. If you do not use the option `--out` no outputs files are generated. The output files can get big very quickly.
```

Comment:
typos: "[...] generates a qubo model that represents [...]", "[...] no output files [...]"